src/main/java/org/victorrobotics/frc/dtlib/command/DTCommandBase.java

```diff
@@ -0,0 +1,50 @@
+package org.victorrobotics.frc.dtlib.command;
+
+import edu.wpi.first.wpilibj2.command.CommandBase;
+
+public abstract class DTCommandBase extends CommandBase {
+    protected abstract void start();
+
+    protected abstract void run();
+
+    protected abstract boolean isComplete();
+
+    protected abstract void finish(boolean interrupted);
+
+    @Override
+    public final void initialize() {
+        try {
+            start();
+        } catch (Exception e) {
+            cancel();
```

Comment:
As we discussed, it would be good to have a way to indicate that a command "failed" and that its parent command group should (usually) also be marked as failed. There should also be a way to stop the propagation of failure up the chain, for example by marking a command as "optional" or by allowing the parent command group to be marked as "continue on failure".

Catching exceptions here is a sound practice and I like having it centralized. Canceling the command is a necessary step but we need to think through whether canceling a command should imply marking it as failed.